src/abs.ott

```diff
@@ -142,6 +140,24 @@ Equations fresh_vars_e (l : list x) (e0 : e) : Prop := {
 where fresh_vars_el (l : list x) (el0 : list e) : Prop := {
  fresh_vars_el l nil := True;
  fresh_vars_el l (e1::el0) := fresh_vars_e l e1 /\ fresh_vars_el l el0 }.
+
+Fixpoint fresh_vars_s (l : list x) (s0 : s): Prop :=
+  match l with
+  | nil => True
+  | (y::ys) => ~ Map.In y s0 /\ fresh_vars_s ys s0
+  end.
+
+Definition fresh_vars (l : list x) (e0: e) (s0: s) : Prop :=
+  fresh_vars_s l s0 /\ fresh_vars_e l e0.
+
+Inductive distinct: list x -> Prop :=
+ | distinct_nil: distinct nil
+ | distinct_cons: forall y ys,
+      ~ In y ys ->
+     distinct ys ->
+     distinct (y::ys).
```

Comment:
Note that the `NoDup` definition from the `List` module in the Coq standard library is isomorphic to the definition of `distinct`. You probably want to switch to `NoDup`: https://coq.inria.fr/doc/master/stdlib/Coq.Lists.List.html#NoDup